application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,111 @@
+import net.dv8tion.jda.api.utils.AttachmentOption;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.TemporalAccessor;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class that allows you to easily log an entry on the mod audit log channel. Thread-Safe.
+ */
+public enum ModAuditLogWriter {
+    ;
+    private static final Color EMBED_COLOR = Color.decode("#3788AC");
+
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static Pattern auditLogChannelNamePattern = null;
+
+    /**
+     * Sends a log embed on the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user to be added to the embed
+     *
+     * @param timestamp the timestamp to be added to the embed
+     *
+     * @param guild the guild to write this log to
+     *
+     * @param attachments attachments that'll be added to the message
+     */
+    public static void write(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull Attachment... attachments) {
+        Optional<TextChannel> auditLogChannel = getAndHandleModAuditLogChannel(guild);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        MessageAction message = auditLogChannel.orElseThrow()
+            .sendMessageEmbeds(new EmbedBuilder().setTitle(title)
+                .setDescription(description)
+                .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                .setTimestamp(timestamp)
+                .setColor(EMBED_COLOR)
+                .build());
+
+        for (Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContentRaw(), attachment.name());
+        }
+        message.queue();
+    }
+
+    /**
+     * Gets the channel used for moderation audit logs, if present. If the channel doesn't exist,
+     * this method will return an empty optional, and a warning message will be written.
+     *
+     * @param guild the guild to look for the channel in
+     */
+    public static Optional<TextChannel> getAndHandleModAuditLogChannel(@NotNull Guild guild) {
+        if (auditLogChannelNamePattern == null) {
+            auditLogChannelNamePattern =
+                    Pattern.compile(Config.getInstance().getModAuditLogChannelPattern());
+        }
+
+        Optional<TextChannel> channel = guild.getTextChannelCache()
+            .stream()
+            .filter(c -> auditLogChannelNamePattern.asMatchPredicate().test(c.getName()))
```

Comment:
what is `c`? bad name. instead `channel`. and rename the other channel variable to `auditLogChannel` to clarify the difference

Reply:
done.